chain/client/src/stateless_validation/partial_witness_tracker.rs

```diff
@@ -0,0 +1,177 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use lru::LruCache;
+use near_chain::Error;
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::reed_solomon::ReedSolomonWrapper;
+use near_primitives::sharding::ChunkHash;
+use near_primitives::stateless_validation::{EncodedChunkStateWitness, PartialEncodedStateWitness};
+
+/// Max number of chunks to keep in the witness tracker cache. We reach here only after validation
+/// of the partial_witness so the LRU cache size need not be too large.
+const NUM_CHUNKS_IN_WITNESS_TRACKER_CACHE: usize = 200;
+
+/// Ratio of the number of data parts to total parts in the Reed Solomon encoding.
+/// The tradeoff here is having a higher ratio is better for handling missing parts and network errors
+/// but increases the size of the encoded state witness and the total network bandwidth requirements.
+const RATIO_DATA_PARTS: f32 = 0.8;
```

Comment:
At the beginning, I'm keeping this ratio as 0.8 instead of 0.66
Let's see how well this works and we can always tweak this or make it a parameter later.